### ISSUE TITLE
Traceback/FileNotFoundError since ae02a77

### ISSUE THREAD

**Issue by KlfJoat** [DIGITALCRIMINALS/OnlyFans#373]
I have an issue that seems similar to #320 but slightly different. More specifically it's like the comment by @adamperez in https://github.com/DIGITALCRIMINALS/OnlyFans/issues/320#issuecomment-1060729449_

Since updating to ae02a773b009cf31ae874d4af57e4b136122002e I've had a double Traceback.

```
...
Scrape Processing
Name: MODELNAME
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/shutil.py", line 805, in move
    os.rename(src, real_dst)
FileNotFoundError: [Errno 2] No such file or directory: '/Pr0n/OnlyFans/MODELNAME/Metadata/Posts.json' -> '.sites/OnlyFans/
MODELNAME/Metadata/Posts.json'
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/src/OnlyFans-working/start_ofd.py", line 66, in <module>
    asyncio.run(main())
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "/src/OnlyFans-working/start_ofd.py", line 52, in main
    api = await main_datascraper.start_datascraper(config, site_name)
  File "/src/OnlyFans-working/datascraper/main_datascraper.py", line 131, in start_datascraper
    await default(datascraper)
  File "/src/OnlyFans-working/datascraper/main_datascraper.py", line 101, in default
    await main_helper.process_jobs(datascraper, subscription_list, site_settings)
  File "/src/OnlyFans-working/helpers/main_helper.py", line 1049, in process_jobs
    await datascraper.start_datascraper(authed, username,whitelist=["Messages"])
  File "/src/OnlyFans-working/modules/module_streamliner.py", line 70, in start_datascraper
    await main_helper.format_directories(
  File "/src/OnlyFans-working/helpers/main_helper.py", line 1391, in format_directories
    shutil.move(metadata_filepath, new_m_f)
  File "/usr/local/lib/python3.10/shutil.py", line 825, in move
    copy_function(src, real_dst)
  File "/usr/local/lib/python3.10/shutil.py", line 434, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/local/lib/python3.10/shutil.py", line 256, in copyfile
    with open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: '.sites/OnlyFans/MODELNAME/Metadata/Posts.json'
```

Since both `/Pr0n/OnlyFans/MODELNAME/Metadata/Posts.json` exists and `.sites/OnlyFans/MODELNAME/Metadata/` exists, I don't think that I'm having #320 since the workaround there already describes my current state.

I notice it's for a model I no longer subscribe to. I notice that the script is now going over all models, not just ones I'm currently subscribed to.  I suspected that it might have something to do with commit https://github.com/DIGITALCRIMINALS/OnlyFans/commit/ae02a773b009cf31ae874d4af57e4b136122002e to grab Messages content for models you're not subscribed to. I say that because of the timing and because rolling back fixes the problem.

I checked out commit 74797996985b6bc1643eb303d6f92dd020f81696 (before the Messages update) and the script ran successfully without issues, without the FileNotFoundError, and without pulling from models I'm not subscribed to anymore.

**Comment by headphonia** [comment]
I'm having the same issue with models I'm no longer subscribed to, there seems to be a bug related to merging the metadata from multiple databases into a single one? I say this because if I append a ".old" to the previous database files it'll run to the models just fine.

`Traceback (most recent call last):
  File "/Users/USER/.pyenv/versions/3.10.0/lib/python3.10/shutil.py", line 805, in move
    os.rename(src, real_dst)
FileNotFoundError: [Errno 2] No such file or directory: '.sites/OnlyFans/OFUSERNAME/Metadata/Archived.db' -> '.sites/OnlyFans/OFUSERNAME/Metadata/__legacy_metadata__/Archived.db'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "start_ofd.py", line 66, in <module>
    asyncio.run(main())
  File "/Users/USER/.pyenv/versions/3.10.0/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/USER/.pyenv/versions/3.10.0/lib/python3.10/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "start_ofd.py", line 52, in main
    api = await main_datascraper.start_datascraper(config, site_name)
  File "datascraper/main_datascraper.py", line 131, in start_datascraper
    await default(datascraper)
  File "datascraper/main_datascraper.py", line 101, in default
    await main_helper.process_jobs(datascraper, subscription_list, site_settings)
  File "helpers/main_helper.py", line 1049, in process_jobs
    await datascraper.start_datascraper(authed, username,whitelist=["Messages"])
  File "modules/module_streamliner.py", line 73, in start_datascraper
    await main_helper.fix_sqlite(api, subscription_directory_manager)
  File "helpers/main_helper.py", line 446, in fix_sqlite
    shutil.move(archived_database_path, f"{new_filepath}")
  File "/Users/USER/.pyenv/versions/3.10.0/lib/python3.10/shutil.py", line 825, in move
    copy_function(src, real_dst)
  File "/Users/USER/.pyenv/versions/3.10.0/lib/python3.10/shutil.py", line 434, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/Users/USER/.pyenv/versions/3.10.0/lib/python3.10/shutil.py", line 256, in copyfile
    with open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: '.sites/OnlyFans/OFUSERNAME/Metadata/__legacy_metadata__/Archived.db'`

I just turned off the "messages" jobs setting, hopefully that will bypass the new code and do a complete run.

**Comment by DIGITALCRIMINALS** [comment]
Is there No Posts.json in the folder?

**Comment by KlfJoat** [comment]
`/Pr0n/OnlyFans/MODELNAME/Metadata/Posts.json` exists. 

Aaand I think I found the problem. I changed my directory structure from `/{model_username}/` to `/{site_name}/{model_username}/`.  So the destination folders didn't exist.

But I see where the script created the destination folders and `{Archived,Posts,Messages}.db` files for a lot of the models. Strange that it worked for a bunch of other models but got stopped on that one model every time. 

Seems like a bug still exists if only some destination folders and files get created, but not all of them. So I'll leave this open.

I restored the ae02a773b009cf31ae874d4af57e4b136122002e commit and ran it and now it seems like I have the same problem as #320. Some models work with no problems, others require that `__legacy_metadata__` destination folder.
